jtelegrambotapi-menus/src/main/java/com/jtelegram/api/menu/impl/BoundMenuImpl.java

```diff
@@ -0,0 +1,47 @@
+package com.jtelegram.api.menu.impl;
+
+import com.jtelegram.api.TelegramBot;
+import com.jtelegram.api.chat.id.ChatId;
+import com.jtelegram.api.menu.BoundMenu;
+import com.jtelegram.api.message.impl.TextMessage;
+import com.jtelegram.api.requests.message.edit.EditTextMessage;
+import javax.annotation.Nonnull;
+
+public class BoundMenuImpl implements BoundMenu {
+
+    private final MenuImpl menu;
+    private final TextMessage message;
+
+    BoundMenuImpl(MenuImpl menu, TextMessage message) {
+        this.menu = menu;
+        this.message = message;
+    }
+
+    @Nonnull
+    @Override
+    public MenuImpl getMenu() {
+        return this.menu;
+    }
+
+    @Nonnull
+    @Override
+    public TextMessage getMessage() {
+        return this.message;
+    }
+
+    @Override
+    public void update(@Nonnull TelegramBot bot) {
+        if (message.getSender().getId() != bot.getBotInfo().getId()) {
+            throw new IllegalArgumentException("Message was not sent by me!");
```

Comment:
This should actually specify which bot it wasn't sent from as this API supports multiple bots operating concurrently